modal/sandbox.py

```diff
@@ -91,7 +91,44 @@ async def _get_logs():
         return data
 
 
+class _StreamWriter:
+    """Provides an interface to buffer and write logs to a sandbox stream (`stdin`)."""
+
+    def __init__(self, sandbox_id: str, client: _Client):
+        self._index = 1
+        self._sandbox_id = sandbox_id
+        self._client = client
+        self._is_closed = False
+        self._buffer = bytearray()
+
+    def get_next_index(self):
+        index = self._index
+        self._index += 1
+        return index
+
+    def write(self, data):
+        if self._is_closed:
+            return
```

Comment:
Shouldn't this be an error? Does writing to a asyncio `StreamWriter` silently drop the write?

Reply:
Yeah I confirmed and looked at their code, they fail silently. But after a certain number of failed writes they log a warning. Maybe we can log a warning for our user?

<img width="585" alt="Screenshot 2024-03-13 at 6 14 18 PM" src="https://github.com/modal-labs/modal-client/assets/47339399/c365cb3e-b90e-44eb-b0d2-a3ad7f1cdad9">

Reply:
Wow that is surprising. There must be some use-case where hard failing 'late' writes doesn't work. I'd opt to be strict until we discover a problem with the strictness.